include/clients/nrt/KDTreeClient.hpp

```diff
@@ -232,9 +232,9 @@ class KDTreeQuery : public FluidBaseClient, ControlIn, ControlOut
       }
 
       auto [dists, ids] =
-          kdtreeptr->algorithm().kNearest(point, k, 0, c.allocator());
+          kdtreeptr->algorithm().kNearest(point, realK, 0, c.allocator()); // i'd like to pass get<kRadius>() and output min(nbpoints,realK) somehow
```

Comment:
Pass the radius as the 3rd argument. Just pass `k` as is, unless I'm missing something subtle. You can find out how many you actually got from the results.